datadog/resource_datadog_synthetics_test_.go

```diff
@@ -2010,11 +2010,15 @@ func buildTestOptions(d *schema.ResourceData) *datadogV1.SyntheticsTestOptions {
 				rumSettings.SetIsEnabled(true)
 
 				if applicationId, ok := settings["application_id"]; ok {
-					rumSettings.SetApplicationId(applicationId.(string))
+					if len(applicationId.(string)) > 0 {
+						rumSettings.SetApplicationId(applicationId.(string))
+					}
 				}
 
 				if clientTokenId, ok := settings["client_token_id"]; ok {
-					rumSettings.SetClientTokenId(int64(clientTokenId.(int)))
+					if clientTokenId.(int) != 0 {
```

Comment:
TF uses `0` when a value hasn't been put, but it is not a valid client token so I think it's safe to check for this.